tests/test_table.py

```diff
@@ -1298,6 +1307,100 @@ def test_add_group_metadata_w_existing_metadata(self):
             {'graph': ('edge_list', '(4,5), (4,6), (5,7), (6,7)'),
              'tree': ('newick', '((4:0.1,5:0.1):0.2,(6:0.1,7:0.1):0.2):0.3;')})
 
+    def test_to_dataframe(self):
+        exp = pd.SparseDataFrame(np.array([[0.0, 1.0, 2.0], [3.0, 4.0, 5.0]]),
+                                 index=['O1', 'O2'],
+                                 columns=['S1', 'S2', 'S3'])
+        obs = example_table.to_dataframe()
+        pdt.assert_frame_equal(obs, exp)
+
+    def test_metadata_to_dataframe(self):
+        exp_samp = pd.DataFrame(['A', 'B', 'A'], index=['S1', 'S2', 'S3'],
+                                columns=['environment'])
+        exp_obs = pd.DataFrame([['Bacteria', 'Firmicutes'],
+                                ['Bacteria', 'Bacteroidetes']],
+                               index=['O1', 'O2'],
+                               columns=['taxonomy_0', 'taxonomy_1'])
+        obs_samp = example_table.metadata_to_dataframe(axis='sample')
+        obs_obs = example_table.metadata_to_dataframe(axis='observation')
+        pdt.assert_frame_equal(obs_samp, exp_samp)
+        pdt.assert_frame_equal(obs_obs, exp_obs)
```

Comment:
Can we have an extra test case for the scenario where the taxonomies aren't complete?

For example, is it possible to return the following dataframe

```
exp_obs = pd.DataFrame([['B', 'F'], ['B'], index=['O1', 'O2'], columns=['taxonomy_0', 'taxonomy_1'])
```
where the phylum level is missing for one of the OTUs